markovic/YahooNewsFeed/app/src/main/java/com/example/aleksandarmarkovic/yahoonewsfeed/NewsListAdapter.java

```diff
@@ -1,50 +1,102 @@
 package com.example.aleksandarmarkovic.yahoonewsfeed;
 
-import android.content.Context;
+import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ArrayAdapter;
+import android.widget.ImageView;
 import android.widget.TextView;
 
 import com.example.aleksandarmarkovic.yahoonewsfeed.database.SingleNewsItem;
 
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Created by aleksandar.markovic on 6/15/2015.
  */
-public class NewsListAdapter extends ArrayAdapter<SingleNewsItem> {
+public class NewsListAdapter extends RecyclerView.Adapter<NewsListAdapter.ViewHolder> {
 
-    private LayoutInflater inflater;
+    NewsItemOnClickListener newsItemOnClickListener;
+    private List<SingleNewsItem> newsItemList;
 
-    public NewsListAdapter(Context context) {
-        super(context, android.R.layout.simple_list_item_2);
-        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+    public NewsListAdapter(NewsItemOnClickListener newsItemOnClickListener) {
+        this.newsItemOnClickListener = newsItemOnClickListener;
+        this.newsItemList = new ArrayList<>();
     }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-        View view;
 
-        if (convertView == null) {
-            view = inflater.inflate(R.layout.single_news_list_item, parent, false);
-        } else {
-            view = convertView;
+    public void setData(List<SingleNewsItem> newData) {
+        newsItemList.clear();
```

Comment:
You will clear list though newData can be null. In this case you will show nothing for user